<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>610</width>
    <height>398</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TyQt</string>
  </property>
  <property name="windowIcon">
   <iconset resource="tyqt.qrc">
    <normaloff>:/tyqt</normaloff>:/tyqt</iconset>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QListView" name="boardList">
       <property name="minimumSize">
        <size>
         <width>180</width>
         <height>0</height>
        </size>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::ExtendedSelection</enum>
       </property>
       <property name="selectionRectVisible">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QTabWidget" name="tabWidget">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="infoTab">
        <attribute name="title">
         <string>&amp;Information</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="1" column="1">
            <widget class="QLineEdit" name="statusText">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Status:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Identity:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLineEdit" name="idText">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="1">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Serial Number:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="serialText">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLineEdit" name="locationText">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Location:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLineEdit" name="modelText">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Model:</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Interfaces:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTreeWidget" name="interfaceTree">
           <property name="indentation">
            <number>0</number>
           </property>
           <property name="rootIsDecorated">
            <bool>false</bool>
           </property>
           <property name="uniformRowHeights">
            <bool>true</bool>
           </property>
           <attribute name="headerDefaultSectionSize">
            <number>100</number>
           </attribute>
           <attribute name="headerMinimumSectionSize">
            <number>100</number>
           </attribute>
           <attribute name="headerStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Path</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="monitorTab">
        <attribute name="title">
         <string>&amp;Monitor</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPlainTextEdit" name="monitorText">
           <property name="minimumSize">
            <size>
             <width>240</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="undoRedoEnabled">
            <bool>false</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLineEdit" name="monitorEdit">
             <property name="placeholderText">
              <string>Send data...</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="sendButton">
             <property name="focusPolicy">
              <enum>Qt::StrongFocus</enum>
             </property>
             <property name="text">
              <string>Send</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="uploadTab">
        <attribute name="title">
         <string>&amp;Settings</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Upload</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Firmware:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="firmwarePath">
                <property name="clearButtonEnabled">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="firmwareBrowseButton">
                <property name="focusPolicy">
                 <enum>Qt::StrongFocus</enum>
                </property>
                <property name="text">
                 <string>Browse...</string>
                </property>
                <property name="popupMode">
                 <enum>QToolButton::MenuButtonPopup</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="resetAfterCheck">
              <property name="text">
               <string>Reset device after successful upload</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Monitor</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QCheckBox" name="clearOnResetCheck">
                <property name="text">
                 <string>Clear on reset</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Scrollback:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="scrollBackLimitSpin">
                <property name="accelerated">
                 <bool>true</bool>
                </property>
                <property name="suffix">
                 <string> lines</string>
                </property>
                <property name="minimum">
                 <number>10</number>
                </property>
                <property name="maximum">
                 <number>500000</number>
                </property>
                <property name="singleStep">
                 <number>100</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QCheckBox" name="echoCheck">
                <property name="text">
                 <string>Echo</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QComboBox" name="newlineComboBox">
                <property name="currentIndex">
                 <number>1</number>
                </property>
                <item>
                 <property name="text">
                  <string>No line ending</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Newline (LF)</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Carriage return (CR)</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Both (CRLF)</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>610</width>
     <height>21</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::PreventContextMenu</enum>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;Actions</string>
    </property>
    <widget class="QMenu" name="menuRecentFirmwares">
     <property name="title">
      <string>Upload recent &amp;firmware</string>
     </property>
    </widget>
    <addaction name="separator"/>
    <addaction name="actionUpload"/>
    <addaction name="actionUploadNew"/>
    <addaction name="menuRecentFirmwares"/>
    <addaction name="separator"/>
    <addaction name="actionReset"/>
    <addaction name="actionReboot"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionWebsite"/>
    <addaction name="actionReportBug"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionNewWindow"/>
    <addaction name="actionMinimalInterface"/>
    <addaction name="separator"/>
    <addaction name="actionClearMonitor"/>
    <addaction name="actionAttachMonitor"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="actionArduinoTool"/>
    <addaction name="separator"/>
    <addaction name="actionResetApp"/>
    <addaction name="actionResetSettingsApp"/>
    <addaction name="separator"/>
    <addaction name="actionOpenLog"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="palette">
    <palette>
     <active>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
     </active>
     <inactive>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
     </inactive>
     <disabled>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>120</red>
         <green>120</green>
         <blue>120</blue>
        </color>
       </brush>
      </colorrole>
     </disabled>
    </palette>
   </property>
   <property name="sizeGripEnabled">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="contextMenuPolicy">
    <enum>Qt::PreventContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionUpload"/>
   <addaction name="actionReset"/>
   <addaction name="actionReboot"/>
   <addaction name="separator"/>
   <addaction name="actionAttachMonitor"/>
  </widget>
  <action name="actionUpload">
   <property name="icon">
    <iconset resource="tyqt.qrc">
     <normaloff>:/upload</normaloff>:/upload</iconset>
   </property>
   <property name="text">
    <string>&amp;Upload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="icon">
    <iconset resource="tyqt.qrc">
     <normaloff>:/reset</normaloff>:/reset</iconset>
   </property>
   <property name="text">
    <string>&amp;Reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionReboot">
   <property name="icon">
    <iconset resource="tyqt.qrc">
     <normaloff>:/reboot</normaloff>:/reboot</iconset>
   </property>
   <property name="text">
    <string>&amp;Bootloader</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionUploadNew">
   <property name="text">
    <string>Upload &amp;New Firmware</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+U</string>
   </property>
  </action>
  <action name="actionWebsite">
   <property name="text">
    <string>&amp;Website</string>
   </property>
  </action>
  <action name="actionReportBug">
   <property name="text">
    <string>&amp;Report a bug</string>
   </property>
  </action>
  <action name="actionMinimalInterface">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Minimal Interface</string>
   </property>
   <property name="toolTip">
    <string>Minimal Interface</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionNewWindow">
   <property name="text">
    <string>&amp;New Window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionClearMonitor">
   <property name="text">
    <string>&amp;Clear Monitor</string>
   </property>
   <property name="toolTip">
    <string>Clear Monitor</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+X</string>
   </property>
  </action>
  <action name="actionArduinoTool">
   <property name="text">
    <string>&amp;Integrate to Arduino</string>
   </property>
  </action>
  <action name="actionAttachMonitor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="tyqt.qrc">
     <normaloff>:/monitor</normaloff>:/monitor</iconset>
   </property>
   <property name="text">
    <string>&amp;Monitor</string>
   </property>
   <property name="toolTip">
    <string>Monitor</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionResetApp">
   <property name="text">
    <string>Reset TyQt</string>
   </property>
  </action>
  <action name="actionResetSettingsApp">
   <property name="text">
    <string>Reset Settings &amp;&amp; TyQt</string>
   </property>
   <property name="toolTip">
    <string>Reset Settings &amp; TyQt</string>
   </property>
  </action>
  <action name="actionOpenLog">
   <property name="text">
    <string>Open &amp;Log</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>boardList</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>idText</tabstop>
  <tabstop>statusText</tabstop>
  <tabstop>modelText</tabstop>
  <tabstop>serialText</tabstop>
  <tabstop>locationText</tabstop>
  <tabstop>interfaceTree</tabstop>
  <tabstop>monitorText</tabstop>
  <tabstop>monitorEdit</tabstop>
  <tabstop>sendButton</tabstop>
  <tabstop>firmwarePath</tabstop>
  <tabstop>firmwareBrowseButton</tabstop>
  <tabstop>resetAfterCheck</tabstop>
  <tabstop>clearOnResetCheck</tabstop>
  <tabstop>scrollBackLimitSpin</tabstop>
  <tabstop>echoCheck</tabstop>
  <tabstop>newlineComboBox</tabstop>
 </tabstops>
 <resources>
  <include location="tyqt.qrc"/>
 </resources>
 <connections/>
</ui>
